### ISSUE TITLE
Add permission nodes for plugin messages

### ISSUE THREAD

**Issue by Phoenix616** [jrbudda/Vivecraft_Spigot_Extensions#50]
Currently anyone can just send a custom plugin message on the VSE plugin message channel and have the server execute any plugin message action with teleporting the player to an arbitrary location probably being the most usefully exploitable one. This could be solved by adding permission nodes, at least one for registering a player as a VivePlayer so that not everyone can just switch to that mode or even one node for each individual plugin message packet type.

**Comment by Techjar** [comment]
We are aware of this, and didn't bother to do anything about it, as if someone really wanted to exploit it they could just lie to server about being a VR player.

**Comment by Phoenix616** [comment]
Yeah which is why I included to at least have a permission for allowing someone to register as a VivePlayer so that this could only be granted to trusted players.

**Comment by Techjar** [comment]
That isn't really user-friendly.

**Comment by Techjar** [comment]
Teleport is designed to be available to as many players as possible, because VR comfort can be a real issue.

**Comment by Phoenix616** [comment]
That is for the server owner/admin to decide though. If they install this plugin and only really want to grant the ability do use it only to certain players then they should be able to do so.

**Comment by Techjar** [comment]
Yeah I guess it's also up to the server admin to decide if they want to install the plugin.

¯\_(ツ)_/¯

**Comment by Techjar** [comment]
I suppose we could just add an option to disable teleport on the server side, as it's probably one of the most "cheaty" aspects of Vivecraft (though it's not particularly advantageous in many cases). I still don't agree with it for comfort reasons, but this would at least be equivalent to not having the plugin, except you still get the other patches to fix quirks with things like Endermen.
